engine/src/main/java/org/hibernate/search/engine/common/resources/impl/EngineThreads.java

```diff
@@ -33,6 +33,9 @@ public ScheduledExecutorService getTimingExecutor() {
 		if ( executor != null ) {
 			return executor;
 		}
+		// this sync block doesn't seem to require a change:
+		// - sync happens only once on init ?
+		// - no I/O and simple in-memory operations
```

Comment:
I agree, but here too I think the comment needs some work.

```suggestion
		// this sync block shouldn't be a problem for Loom:
		// - sync happens only once on init ?
		// - no I/O and simple in-memory operations
```

Reply:
Also, you might want to add the same comment to `org.hibernate.search.engine.common.timing.impl.DefaultTimingSource#ensureTimeEstimateIsInitialized`.